src/pages/index.js

```diff
@@ -37,35 +104,85 @@ function renderCards(element, elementList) {
 
 const cardList = new Section(
   {
-    items: getInitialCards(),
+    items: [],
     renderer: renderCards,
   },
   cardListLocation
 );
-cardList.renderItems();
 
-//////////////////// Profile Edit Form
+// API
+function setInitialUserInfo(res) {
+  return res.then((json) => {
+    profileInfo.setUserInfo({
+      name: json.name,
+      description: json.about,
+    });
+    profileInfo.setUserAvatar({ link: json.avatar });
+  });
+}
 
-const profile = document.querySelector(".profile");
-const profileName = profile.querySelector(".profile__name");
-const profileDecsription = profile.querySelector(".profile__description");
+function setInitialCards(res) {
+  return res.then((json) => {
+    const cards = [];
+    json.forEach((card) => {
+      const cardElement = createCard(card);
+      cards.unshift(cardElement);
+    });
+    cardList.setItems(cards);
+    cardList.renderItems();
+  });
+}
 
-const profileInfo = new UserInfo({
-  name: profileName,
-  description: profileDecsription,
-});
+const api = new Api({ token: token });
+api
+  .initialize()
+  .then((resArr) => {
+    if (resArr[0].ok && resArr[1].ok) {
+      return (
+        setInitialUserInfo(resArr[0].json()) + setInitialCards(resArr[1].json())
+      );
+    } else {
+      return Promise.reject(
+        `Failed to Initialize: User Info Status: ${resArr[0].status} Card List Status: ${resArr[1].status}`
+      );
+    }
+  })
+  .catch((err) => {
+    console.error(`ERROR: ${err}`);
+  });
+
+//////////////////// Profile Edit Form
 
 function handleProfileFormSubmit(inputValues) {
-  profileInfo.setUserInfo(inputValues);
-  profileEditModal.close();
+  api
+    .updateUserInfo(inputValues)
+    .then((res) => {
+      if (res.ok) {
+        return res.json();
+      } else {
+        return Promise.reject(
+          `Error: ${res.status} Couldn't update profile info`
+        );
+      }
+    })
+    .then((res) => {
+      profileInfo.setUserInfo({ name: res.name, description: res.about });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditModal.close();
+      profileEditModal.resetSubmitButtonText();
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```